lib/hyacinth/datacite/api_session.rb

```diff
@@ -0,0 +1,117 @@
+        @last_response_from_server = Hyacinth::Datacite::ServerResponse.new response
+        Hyacinth::Utils::Logger.logger.error("#mint_identifier: Response from DataCite server failed:\n" \
+                                             "doi_status: #{doi_status}\n" \
+                                             "target_url: #{target_url}\n" \
+                                             "url suffix: #{uri}\n" \
+                                             "datacite attributes: #{attributes_hash.inspect}\n" \
+                                             "response body: #{response.body}") if @last_response_from_server.error?
+        Hyacinth::Datacite::Doi.new(@last_response_from_server.doi,
+                                    doi_status) unless @last_response_from_server.error?
+      rescue Net::ReadTimeout, SocketError
+        Hyacinth::Utils::Logger.logger.error('#mint_identifier: EZID API call to mint identifier timed-out')
+        @timed_out = true
+        nil
+      end
+    end
+
+    def modify_identifier(doi,
+                          doi_status,
+                          attributes_hash,
+                          target_url = nil)
+      attributes_hash[:url] = target_url unless target_url.nil?
+      attributes_hash[:event] = DOI_MINT_EVENT[doi_status] unless doi_status == :draft
+      attributes_hash[:prefix] = EZID[:prefix]
+      uri_path = '/dois/' + doi
+      uri = URI(EZID[:url] + uri_path)
+      request = Net::HTTP::Put.new uri.request_uri
+      begin
+        response = call_api(uri, request, attributes_hash)
+        @last_response_from_server = Hyacinth::Datacite::ServerResponse.new response
+        Hyacinth::Utils::Logger.logger.error("#modify_identifier: Response from DataCite server failed:\n" \
+                                             "doi_status: #{doi_status}\n" \
+                                             "target_url: #{target_url}\n" \
+                                             "url suffix: #{uri}\n" \
+                                             "datacite attributes: #{attributes_hash.inspect}\n" \
+                                             "response body: #{response.body}") if @last_response_from_server.error?
+        Hyacinth::Datacite::Doi.new(@last_response_from_server.doi,
+                                doi_status) unless @last_response_from_server.error?
+      rescue Net::ReadTimeout, SocketError
+        Hyacinth::Utils::Logger.logger.error('modify_identifier: EZID API call to modify identifier timed-out')
+        @timed_out = true
+        false
+      end
+    end
+
+    private
+
+      def call_api(uri, request, attributes = nil)
+        data = { type: 'dois' }
+        data[:attributes] = attributes
+        request.body = { data: data }.to_json
+        # fcd1, 12/23/21: Following can be uncommented for in situ debugging
+        # Hyacinth::Utils::Logger.logger.error("URL: #{uri}")
+        # Hyacinth::Utils::Logger.logger.error("REQUEST BODY #{request.body}")
```

Comment:
Lines like this make sense during active development, but I'd prefer to remove them once the code is production ready.

Reply:
Comments have been removed.